src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntryValidator.cs

```
@@ -0,0 +1,170 @@
+                    // When VR is missing for standard tag, still need to verify VRCode
+                    string vrCode = tag.DictionaryEntry.ValueRepresentations[0].Code;
+                    EnsureVRIsSupported(vrCode);
+                }
+                else
+                {
+                    // when VR is specified, verify it's correct
+                    // parse VR
+                    DicomVR vr = ParseVRCode(tagEntry.VR);
+
+                    EnsureVRIsSupported(vr.Code);
+
+                    if (!tag.DictionaryEntry.ValueRepresentations.Contains(vr))
+                    {
+                        // not a valid VR
+                        throw new CustomTagEntryValidationException(
+                            string.Format(CultureInfo.InvariantCulture, DicomCoreResource.UnsupportedVRCodeOnTag, vr.Code, tag));
+                    }
+                }
+            }
+        }
+
+        private static DicomVR ParseVRCode(string vrCode)
+        {
+            if (string.IsNullOrWhiteSpace(vrCode))
+            {
+                throw new CustomTagEntryValidationException(DicomCoreResource.MissingVRCode);
+            }
+
+            try
+            {
+                // DicomVR.Parse only accept upper case  VR code.
+                return DicomVR.Parse(vrCode.ToUpper(CultureInfo.InvariantCulture));
+            }
+            catch (Exception ex)
+            {
+                throw new CustomTagEntryValidationException(
+                    string.Format(CultureInfo.InvariantCulture, DicomCoreResource.InvalidVRCode, vrCode), ex);
+            }
+        }
+
+        private static DicomTag ParseTag(string path)
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                throw new CustomTagEntryValidationException(DicomCoreResource.MissingCustomTag);
+            }
+
+            try
+            {
+                return DicomTag.Parse(path);
+            }
+            catch (Exception ex)
```

Can we not catch all exceptions here. You will get DicomDataException if parse fails

---

Generally coding practice, don't catch system.exception. Some of the system.exceptions like stackoverflow should not be caught.